### PR TITLE
docs: :memo: add why conventional branches post

### LINKED ISSUE #91: Add a decision post on using Conventional Branches

### PR DESCRIPTION
**Pull request by signekb**
## Description

- This PR adds a post on the naming scheme for branches.
- Since I couldn't find specific, different conventions I chose to go into details on what to include/exclude in a branch name. 

## Related Issues

Closes seedcase-project/team#91

## Checklist

For general documentation:

- [X] Spell-check
- [X] Did the page(s) preview correctly on your machine without breaking
- [ ] New category words (keywords) (if any) added to the code snippet file

### CODE REVIEW

File: why-conventional-branches.qmd [hunk 1]
```diff
@@ -0,0 +1,241 @@
...
+:::
+
+## Context and problem statement
+
+::: content-hidden
+State the context and some background on the issue, then write a
+statement in the form of a question for the problem.
+:::
+
+For the Seedcase project, we aim to follow best practices and have
```
Comment (lwjohnst86) [inline why-conventional-branches.qmd:26]:
```suggestion
For the Seedcase Project, we aim to follow best practices and have
```

File: why-conventional-branches.qmd [hunk 2]
```diff
@@ -0,0 +1,241 @@
...
+
+Like with Conventional Commits, the type prefix enables you to quickly
+identify the purpose of the branch. Prefixes include `feat`, `fix`,
+`docs`, and so on, following the conventional commits convention.
+
+In general, the following best practices apply for branch naming:
+
+1.  Only use lower casing
+
+2.  Only use alphanumeric characters (avoid punctuation, spaces,
```
Comment (lwjohnst86) [inline why-conventional-branches.qmd:92]:
```suggestion
2.  Only use alphanumeric characters (don't use punctuation, spaces,
```

File: why-conventional-branches.qmd [hunk 3]
```diff
@@ -0,0 +1,241 @@
...
+-   Adds complexity to the branch name which could lead to errors and,
+    thus, inconsistencies
+:::
+:::
+
+### Include user ID or username
+
+Some teams include the initials of the author in the branch name to keep
+track of each developer's work. Let's outline the benefits and drawbacks
+of this practice:
```
Comment (lwjohnst86) [inline why-conventional-branches.qmd:171]:
```suggestion
track of each developer's work.
```

**Review by lwjohnst86 — CHANGES_REQUESTED** [review]
Nice! Only additional minor comment is, were there any external links you used that could be included here for reference?

**Comment by signekb** [comment]
> Nice! Only additional minor comment is, were there any external links you used that could be included here for reference?

Definitely! I'll see if I can find them again 🌴

**Review by K-Beicher — DISMISSED** [review]
This is very helpful

**Review by lwjohnst86 — APPROVED** [review]
⭐

**Pull request merged**